### ISSUE TITLE
Support for Rails 7.2

### ISSUE THREAD

**Issue by dwarner** [rsim/oracle-enhanced#2411]
I know 7.1.0 just got released with support for Rails 7.1.x (thanks), but is there an ETA for Rails 7.2 support?

https://github.com/rsim/oracle-enhanced/blob/v7.1.0/activerecord-oracle_enhanced-adapter.gemspec
`s.add_runtime_dependency("activerecord", ["~> 7.1.0"])`

I know Rails 7.2 ActiveRecord has a lot of "Remove deprecated" . . .
https://guides.rubyonrails.org/7_2_release_notes.html#active-record

**Comment by numbcoder** [comment]
+1

**Comment by matthewkrom** [comment]
+1

**Comment by rjschof** [comment]
+1. Any updates on this or any sort of plans to maintain this gem?

**Comment by andynu** [comment]
I started, and got stuck. https://github.com/rsim/oracle-enhanced/pull/2422
Anyone willing to give my very limited progress a look, I'd appreciate additional perspectives and/or suggestions for diagnosing.

**Comment by andynu** [comment]
I've made further progress and cut a new PR for consideration. Not all the way there, but a significant step closer. There are now only four remaining test failures locally.  #2424

**Comment by dwarner** [comment]
@andynu Thanks for your hard work!

We've pulled your 7.2 branch into one of our projects and have NOT run into a problem.

We'll be trying it on other projects coming up.

Do you have an ETA on an official release?

**Comment by andynu** [comment]
Hi Dennis, I'm glad the patch has been working well for you, and me too!
Alas, I am just a random user; I have no access to make releases.

**Comment by akostadinov** [comment]
Whoever tested this, maybe you add an approval so hopefully author attention is captured.

**Comment by joshRpowell** [comment]
> [@andynu](https://github.com/andynu) Thanks for your hard work!
> 
> We've pulled your 7.2 branch into one of our projects and have NOT run into a problem.
> 
> We'll be trying it on other projects coming up.
> 
> Do you have an ETA on an official release?

+1 branch has been running in app ~6mo now. No issues. 
Thanks for all your work @andynu 🥳 
(unfortunately, not able to approve)

**Comment by dwarner** [comment]
@andynu I will make a comment on the PR. 

Have you gotten any responses/feedback from the maintainers?

**Comment by dwarner** [comment]
@rsim Are you still maintaining this project?

**Comment by smarks-1-ND** [comment]
Yes, thank you @andynu . I am testing your branch now and things are looking good. Hoping this can be merged soon.

**Comment by Bjoernsen** [comment]
Thank you @andynu ! I have a quite complex gem that connects to different databases.

Most of my tests worked out of the box with 7.2. Only one test failed. I am not sure if this is a Rails 7.2 behavior, a bug in the adapter, or if I do not know what I am doing.

In my gem, I have a method calling something like
```ruby
SomeValue.where('value BETWEEN ? and ?', min, max)
```

The values for `min` and `max` come from another DB call. In 7.1 it works, in 7.2 I get the error `ActiveRecord::StatementInvalid: OCIError: ORA-01722: invalid number`

The issue is that `min` and `max` can be a `BigDecimal` (e.g. `0.23255e3`).

The solution was to add `.to_f` to the variables.

```ruby
SomeValue.where('value BETWEEN ? and ?', min.to_f, max.to_f)
```

**Comment by net1957** [comment]
Hello,
I tried @andynu fork on my little application connected to an old Oracle 11 DB.
It worked without any problem after the update of the application to  Rails  7.2.2.1.

Hope that a new release come out soon! 
Cheers

**Comment by Bjoernsen** [comment]
I upgraded one of my biggest Rails app (32,000 LOC) to 7.2 using your branch. All tests succeeded and up to now, I cannot find anything, that does not work.

**Comment by matthewkrom** [comment]
Hi all — pinging this thread since it's been two weeks. 

Also, security support for Rails 7.1 (the highest version supported by this gem, aside from forks) ends in 3 months. "Active Support" by the rails team actually ended 3 months ago.

**Comment by akostadinov** [comment]
If a few contributors agree, maybe we can make a fork (hopefully temporary until @rsim returns). I say a few so we are not stuck again with a single point of contact and have a basic form of peer review before merging.

**Comment by matthewkrom** [comment]
We've added 7.2.0.rc1 to a test environment. So far, no issues to report.

**Comment by yahonda** [comment]
Oracle enhanced adapter 7.2.0 has been released.
Thanks for your patience. https://rubygems.org/gems/activerecord-oracle_enhanced-adapter/versions/7.2.0

**Comment by akostadinov** [comment]
Anybody interested in future Rails support, please take a look at #2471 and provide feedback, sorry for the spam but I think it deserves some wider attention.
